src/Stratis.Bitcoin.IntegrationTests/ProofOfStakeSteps.cs

```diff
@@ -57,31 +62,50 @@ public void MineGenesisAndPremineBlocks()
 
         public void MineCoinsToMaturity()
         {
-            this.nodes[this.PremineNode].GenerateStratisWithMiner(100);
+            this.nodes[this.PremineNode].GenerateStratisWithMiner(Convert.ToInt32(this.nodes[this.PremineNode].FullNode.Network.Consensus.CoinbaseMaturity));
             this.sharedSteps.WaitForNodeToSync(this.nodes[this.PremineNode]);
         }
 
         public void PremineNodeMinesTenBlocksMoreEnsuringTheyCanBeStaked()
         {
-            this.nodes[this.PremineNode].GenerateStratisWithMiner(Convert.ToInt32(this.nodes[this.PremineNode].FullNode.Network.Consensus.CoinbaseMaturity));
+            this.nodes[this.PremineNode].GenerateStratisWithMiner(10);
         }
 
         public void PremineNodeStartsStaking()
         {
+            // Get set of transaction IDs present in wallet before staking is started.
+            this.transactionsBeforeStaking.Clear();
+            foreach (TransactionData transactionData in this.nodes[this.PremineNode].FullNode.WalletManager().Wallets
+                .First()
+                .GetAllTransactionsByCoinType((CoinType)this.nodes[this.PremineNode].FullNode.Network.Consensus
+                    .CoinType))
+            {
+                this.transactionsBeforeStaking.Add(transactionData.Id);
+            }
+
             var minter = this.nodes[this.PremineNode].FullNode.NodeService<IPosMinting>();
             minter.Stake(new WalletSecret() { WalletName = PremineWallet, WalletPassword = PremineWalletPassword });
         }
 
         public void PremineNodeWalletHasEarnedCoinsThroughStaking()
         {
-            var network = this.nodes[this.PremineNode].FullNode.Network;
-            var premine = network.Consensus.ProofOfWorkReward + network.Consensus.PremineReward;
-            var mineToMaturity = network.Consensus.ProofOfWorkReward * 110;
-            var balanceShouldBe = premine + mineToMaturity;
+            // If new transactions are appearing in the wallet, staking has been successful. Due to coin maturity settings the
+            // spendable balance of the wallet actually drops after staking, so the wallet balance should not be used to
+            // determine whether staking occurred.
             TestHelper.WaitLoop(() =>
             {
-                long staked = this.nodes[this.PremineNode].FullNode.WalletManager().GetSpendableTransactionsInWallet(this.PremineWallet).Sum(s => s.Transaction.Amount);
-                return staked > balanceShouldBe;
+                foreach (TransactionData transactionData in this.nodes[this.PremineNode].FullNode.WalletManager().Wallets
+                    .First()
+                    .GetAllTransactionsByCoinType((CoinType)this.nodes[this.PremineNode].FullNode.Network.Consensus
+                        .CoinType))
+                {
+                    if (!this.transactionsBeforeStaking.Contains(transactionData.Id))
```

Comment:
@zeptin OK I understand now why this change is required. As a secondary measure, can we perhaps check the coinstake flag on the transactionData to verify that it IS in fact a coinstake tx? We can also check the value so that it matches the network's stake reward?

The whole point of this test is to ensure that we can stake and mint new coins, so just checking that there is a new tx might not be entirely sufficient?

Reply:
Yeah, that seems sensible, I've added a check for that.

Checking the value might be overkill, but it isn't hard to do. It is however dependent on how many outputs the coinstake consists of (I think it is 8 by default, but this can be overridden and could change in future).